File: java/oneapi/src/org/gsm/oneapi/responsebean/PolicyException.java
package org.gsm.oneapi.responsebean;

/**
 * specific error case indicated by the OneAPI server as a Policy Exception
 */
public class PolicyException implements java.io.Serializable {
	private static final long serialVersionUID = -6032985525466832920L;
	
	/**
	 * the distinctive error message identifier
	 */
	private String messageId=null;
	/**
	 * the textual representation of the error
	 */
	private String text=null;
	/**
	 * any instance specific error variables
	 */
	private String[] variables=null;
	
	/**
	 * return the distinctive error message identifier
	 */
	public String getMessageId() { return messageId; }
	/**
	 * return the textual representation of the error
	 */
	public String getText() { return text; }
	/**
	 * return any instance specific error variables
	 */
	public String[] getVariables() { return variables; }
	
	/**
	 * set the distinctive error message identifier. This is called internally to set the contents according to the JSON response. 
	 */
	public void setMessageId(String messageId) { this.messageId=messageId; }
	/**
	 * set the textual representation of the error. This is called internally to set the contents according to the JSON response. 
	 */
	public void setText(String text) { this.text=text; }
	/**
	 * set any instance specific error variables. This is called internally to set the contents according to the JSON response. 
	 */
	public void setVariables(String[] variables) { this.variables=variables; }
	
	/**
	 * default constructor
	 */
	public PolicyException() {			
	}
	
	/**
	 * utility constructor to create a ServiceException object with all fields set
	 * @param messageId
	 * @param text
	 * @param variables
	 */
	public PolicyException(String messageId, String text, String[] variables) {
		this.messageId=messageId;
		this.text=text;
		this.variables=variables;			
	}
		
	/** 
	 * generate a textual representation of the ServiceException instance  
	 */
	public String toString() {
		StringBuffer buffer=new StringBuffer();
		buffer.append("messageId = ");
		buffer.append(messageId);
		buffer.append(", text = ");
		buffer.append(text);
		buffer.append(", variables = {");
		if (variables!=null) {
			for (int i=0; i<variables.length; i++) {
				buffer.append("[");
				buffer.append(i);
				buffer.append("] = ");
				buffer.append(variables[i]);
			}
		}
		buffer.append("}");
		return buffer.toString();		
	}
	
	
}

File: Python/response/PolicyException.py

class PolicyException:
	"""defines a policy exception class - returned by the OneAPI server"""
	
	def __init__(self):
		"""Default class constructor"""
		self.messageId=None
		self.text=None
		self.variables=None
	
	def __init__(self, jsondict):
		"""Class constructor that will create an instance initialised from a parsed JSON data block"""
		self.messageId=None
		if jsondict is not None and 'messageId' in jsondict and jsondict['messageId'] is not None:
			self.messageId=jsondict['messageId']
		self.text=None
		if jsondict is not None and 'text' in jsondict and jsondict['text'] is not None:
			self.text=jsondict['text']
		self.variables=None
		if jsondict is not None and 'variables' in jsondict and jsondict['variables'] is not None:
			self.variables=list()
			fieldValue=jsondict['variables']
			if isinstance(fieldValue,list):
				for item in fieldValue:
					self.variables.append (item)
			else:
				self.variables.append (fieldValue)
	
	def getMessageId(self):
		"""getter for messageId : identifying message Id"""
		return self.messageId
	
	def setMessageId(self,messageId):
		"""setter for messageId : identifying message Id"""
		self.messageId=messageId
	
	def getText(self):
		"""getter for text : textual version of the error message"""
		return self.text
	
	def setText(self,text):
		"""setter for text : textual version of the error message"""
		self.text=text
	
	def getVariables(self):
		"""getter for variables : variable information to return"""
		return self.variables
	
	def setVariables(self,variables):
		"""setter for variables : variable information to return"""
		self.variables=variables
	
